Add accepts_nested_attributes_for support for delegated_type

xtr3me opened the pull request:
### Summary

This PR adds `nested_attributes_for` support to `delegated_type`, this allows a developer to create and update records easily without needing to write specific methods like:

```ruby
class Entry < ApplicationRecord
  delegated_type :entryable, types: %w[ Message Comment ]

  def self.create_with_comment(content, creator: Current.user)
    create! entryable: Comment.new(content: content), creator: creator
  end
end
```

Using `nested_attributes_for` allows to execute the following:

```ruby
class Entry < ApplicationRecord
  delegated_type :entryable, types: %w[ Message Comment ]
  accepts_nested_attributes_for :entryable
end

params = { entry: { entryable_type: 'Comment', entryable_attributes: { content: 'Smiling' } } }
entry = Entry.create(params[:entry])
```

### Why

Nested forms based on `accepts_nested_attributes_for` are very powerful and this is the last piece missing to also be able to use it on Delegated Types.

### Question

Since this is a polymorphic `belongs_to` relationship, what other tests would we like to have? As it is already tested in `TestNestedAttributesOnABelongsToAssociation`

Commit:
Add accepts_nested_attributes_for support when using delegated_type

Co-Authored-By: Zachary Scott <zzakscott@gmail.com>

zzak commented on activerecord/CHANGELOG.md:
```diff
@@ -1,3 +1,35 @@
+
+    *Sjors Baltus*
+
+
```
```suggestion
```

zzak commented on activerecord/lib/active_record/delegated_type.rb:
```diff
@@ -136,6 +136,16 @@ module ActiveRecord
   #
   # Now you can list a bunch of entries, call +Entry#title+, and polymorphism will provide you with the answer.
+  #
+  # == Active Record Nested Attributes
```
```suggestion
  # == Nested Attributes
```

zzak commented on activerecord/lib/active_record/delegated_type.rb:
```diff
@@ -136,6 +136,16 @@ module ActiveRecord
+  #
+  # Enabling nested attributes on a delegated_type association allows you to
+  # create the entry and message in one go:
+  #
```
```suggestion
  #
  #    class Entry < ApplicationRecord
  #      delegated_type :entryable, types: %w[ Message Comment ]
  #      accepts_nested_attributes_for :entryable
  #    end
  #
```

The spacing might be a little off, but you get the idea. Let's add the example which includes `accepts_nested_attributes_for`

zzak commented:
@xtr3me Could you rebase too please? I think this PR could use some more tests, but I also wouldn't put more work into this without getting approval on the base of the feature from another committer 🙏

xtr3me commented:
Hi @zzak i have incorporated your suggestions and rebased with main, thank you for reviewing it 👍 
I was wondering of you or anyone else for that matter could guide me in which tests are required extra?

zzak commented:
@xtr3me Sorry for the late reply, could you squash your commits too please?

xtr3me commented:
@zzak i've squashed the commits. Thank you very much for reviewing it.

xtr3me commented:
@zzak I have rebased the branch with main. Is there anything else I could do to advance this into merging it in main?

zzak commented:
Thanks for your patience @xtr3me I've pretty limited capacity right now, so hopefully another review can take a look 🙏

xtr3me commented:
@dhh as the original composer of the delegated_types feature could you maybe help out by reviewing this PR and guide me on which changes are needed to get this merged in?

Or do you know somebody who can/has some spare time to review this PR? 👍

dhh commented:
This is lovely. Thanks for working on it!

0215526 commented:
Loved it 😍

Piioo commented:
How to change from Message to Comment? (with removing the Message from DB)
